tests/scalability/json_generator.py

```diff
@@ -0,0 +1,273 @@
+        return impostor
+
+    def _string(self, schema: dict) -> str:
+        impostor = self._common(schema)
+        if impostor is None:
+            generate_format = schema.get('format')
+            generate_pattern = schema.get('pattern')
+            if generate_format:
+                impostor = getattr(self.faker, self.formats[generate_format])()
+            elif generate_pattern:
+                impostor = rstr.xeger(generate_pattern)
+            else:
+                maximum = schema.get('maxLength', self.UNBOUND_MAX_STRING)
+                minimum = schema.get('minLength', self.UNBOUND_MIN_STRING)
+                impostor = self.faker.pystr(minimum, maximum)
+        return impostor
+
+    def _boolean(self, schema: dict) -> bool:
+        impostor = self._common(schema)
+        if impostor is None:
+            impostor = self.faker.pybool()
+        return impostor
+
+    def _array(self, schema: dict) -> list:
+        items = schema.get('items')
+        contains = schema.get('contains')
+        unique = schema.get('uniqueItems', False)
+        minimum = schema.get('minItems', self.UNBOUND_MIN_ITEMS)
+        maximum = schema.get('maxItems', minimum + self.UNBOUND_MAX_ITEMS)
+        length = minimum if minimum == maximum else random.randrange(minimum, maximum)
+
+        def simple_gen(make_item):
+            if enums:
+                impostor.extend(random.choices(enums, k=length))
+            else:
+                while len(impostor) < length:
+                    impostor.append(self._gen_json(make_item))
+
+        def unique_gen(make_item):
+            if enums:
+                random.shuffle(enums)
+                impostor.extend(enums[:length])
+            else:
+                _retry = 3   # To prevent infinite loops
+                while len(impostor) < length and _retry:
+                    _retry -= 1
+                    u_item = self._gen_json(make_item)
+                    if u_item not in impostor:
+                        impostor.append(u_item)
+                        _retry = 3
+
+        impostor = schema.get('const', [])
+        enums = schema.get('enum')
+        if isinstance(items, dict):
+            if contains:
+                impostor.append(self._gen_json(contains))
+            if unique:
+                unique_gen(items)
+            else:
+                simple_gen(items)
+        elif isinstance(items, list):
+            # TODO: Handle contains
+            additional_items = schema.get('additionalItems')
+            if unique:
+                item_count = len(items)
+                i = 0
+                retry = 3  # To prevent infinite loops
+                while i < item_count and retry:
+                    retry -= 1
+                    item = self._gen_json(items[i])
+                    if item not in impostor:
+                        impostor.append(item)
+                        i += 1
+                        retry = 3
```

Comment:
It's a bit brittle that this line duplicates the initial value for retry above the loop. Is it possible to make retry start at zero, be incremented instead of decremented and place the limit in the loop condition?

Reply:
Are you ok with getting fewer items than item_count in case the retries are exhausted. I guess you are since this is supposed to be random anyways.

Reply:
The chance of getting a duplicate is pretty low to begin with. However there is a chance that minItems will not be satisfied in which case validation could would fail. Since we are producing random test data I think producing some invalid data is helpful for testing. It's worth revisiting in the future if we continue to use this solution.